### ISSUE TITLE
Broken OpenGL

### ISSUE THREAD

**Issue by SevNeutral** [Lactozilla/SRB2-Android#11]
![Screenshot_20220923-081429](https://user-images.githubusercontent.com/114208882/191896140-ab444500-eb2d-49f7-833a-bf24d034272f.jpeg)
![Screenshot_20220923-081547](https://user-images.githubusercontent.com/114208882/191896158-16628b43-e072-4f07-b40d-b2ffba0e6fa4.jpeg)
![Screenshot_20220923-081603](https://user-images.githubusercontent.com/114208882/191896171-6309ccd8-0929-459f-8354-73ae9d7eeece.jpeg)

SRB2 Works incorrectly, there is no skybox, some objects. On software everything is displayed correctly. my phone is "F150 R2022"

**Comment by Lactozilla** [comment]
Sky dome problem seems recurrent. Likely an obscure shader issue?

**Comment by SevNeutral** [comment]
![Screenshot_20220925-114858](https://user-images.githubusercontent.com/114208882/192135535-f76bc413-eaed-4261-a082-7dccd96a2604.jpeg)

**Comment by ashifolfi** [comment]
@Lactozilla I would like to add that this occurs on my Samsung Galaxy Tab S6 Lite, but not on my Galaxy S10e. The only real difference being the tablet uses an Exynos chip while the phone uses a Qualcomm.

**Comment by AndrewtheAndroiduser** [comment]
![srb20038](https://github.com/SRB2-Mobile/SRB2-Android/assets/134543326/c3f4f6ee-a886-474a-8874-0ceeb4bb485d)
So one of the options in opengl settings caused this

**Comment by Cosmiinox** [comment]
I also have this issue, fades also are broken, seems to fix itself by having models loaded, and toggling ambient lightning inside a stage

**Comment by bitten2up** [comment]
on the discord someone solved it by disabling `gr_skydome`

**Comment by Cosmiinox** [comment]
> on the discord someone solved it by disabling `gr_skydome`

Yeah, i knew it before having a github account, also rn i'm not using discord

**Comment by ffoxD** [comment]
i also have the broken skydome and fade screens.
i have a Mali-G57 MC2 GPU

**Comment by Cosmiinox** [comment]
https://youtu.be/0N7m6FmEs40

i made a fix in video form (no audio since srb2 doesnt let me get it by using internal audio in my screen recorder's settings)

also when i messed with the other settings, they were accidents

**Comment by TransBluelight** [comment]
I have a theory and I'm guessing the skydome breaks as it probably uses the GL_OES_framebuffer_object extension to render, this extension is present on Adreno GPUs but is missing for Mali GPUs, the sky renders fine for me but that's because I have an Adreno GPU, specifically an Adreno 610
![srb20015](https://github.com/SRB2-Mobile/SRB2-Android/assets/69445885/1a45cbce-0dfe-4730-a137-d6de03285330)
